app/src/org/commcare/dalvik/activities/CommCareSetupActivity.java

```diff
@@ -550,65 +495,31 @@ public void done(boolean requireRefresh) {
             i.putExtra(KEY_REQUIRE_REFRESH, requireRefresh);
             startActivity(i);
             finish();
-            
-            return;
         } else {
             //Good to go
             Intent i = new Intent(getIntent());
             i.putExtra(KEY_REQUIRE_REFRESH, requireRefresh);
             setResult(RESULT_OK, i);
             finish();
-            return;
         }
     }
 
-    public void fail(NotificationMessage message) {
-        fail(message, false);
-    }
-    
-    public void fail(NotificationMessage message, boolean alwaysNotify) {    
-        fail(message, alwaysNotify, true);
-    }
-    
-    public void fail(NotificationMessage message, boolean alwaysNotify, boolean canRetry){
-
+    void fail(NotificationMessage message, boolean alwaysNotify) {
         Toast.makeText(this, message.getTitle(), Toast.LENGTH_LONG).show();
         
-        retryCount++;
-        
-        if(retryCount > RETRY_LIMIT){
-            canRetry = false;
-        }
-        
-        if(isAuto || alwaysNotify) {
+        if (isAuto || alwaysNotify) {
             CommCareApplication._().reportNotificationMessage(message);
-        }
-        if(isAuto) {
             done(false);
-        } else {
-            if(alwaysNotify) {
-                this.displayMessage= Localization.get("notification.for.details.setup.wrapper", new String[] {message.getDetails()});
-                this.canRetry = canRetry;
-            } else {
-                
-                this.displayMessage= message.getDetails();
-                this.canRetry = canRetry;
-                
-                String fullErrorMessage = message.getDetails();
-                
-                if(alwaysNotify){
-                    fullErrorMessage = fullErrorMessage + message.getAction();

```

Comment:
All of this logic was useless because the vars being assigned are never read elsewhere